src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -0,0 +1,417 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.ComposeCache.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved. 
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--
+    ============================================================
+                                        ComposeCache
+ 
+    The main cache entry point.
+    ============================================================
+    -->
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+
+  <Target Name="ComposeCache"
+          DependsOnTargets="PrepareForComposeCache;
+                            Restore;
+                            ComputeAndCopyFilesToCacheDirectory"/>
+
+  <!--
+    ============================================================
+                                        PrepareForComposeCache
+
+    Prepare the prerequisites for ComposeCache.
+    ============================================================
+    -->
+  <Target Name="PrepareForComposeCache">
+
+    <PropertyGroup>
+      <PreserveCacheLayout Condition="'$(PreserveCacheLayout)' == ''">true</PreserveCacheLayout>
+      <PublishDir>$(ComposeDir)</PublishDir>
+      <MSBUILDENABLEALLPROPERTYFUNCTIONS>1</MSBUILDENABLEALLPROPERTYFUNCTIONS>
```

Comment:
I am setting this msbuild property so that [System.Environment]::get_OSVersion() works, this function is not there in the msbuild's whitelist of available static functions.